server/src/lib.rs

```diff
@@ -318,7 +315,7 @@ async fn mirror(
         op_move_response: &op_move_response,
         port,
     };
-    println!("{}", serde_json::to_string_pretty(&log).unwrap());
+    println!("{}", serde_json::to_string(&log).unwrap());
```

Comment:
Should we consider moving to something like [`tracing`](https://docs.rs/tracing/latest/tracing/) eventually? Then we could have different sinks for jsonified or prettified log output, for example. Among other observability knick-knacks.